src/mutableAtom/index.ts

```diff
@@ -30,31 +30,28 @@ export function makeMutableAtom<Value>(
 
   const storeAtom = atom<
     Store<Value>,
-    [ActionWithPayload<'setValueAsync', Value> | Action<'mount'>],
-    void
+    [ActionWithPayload<'setValue', Value> | Action<'getValue'>],
+    void | Value
   >(
-    (get, { setSelf }) => ({
-      proxyState: null,
-      getValue: () => get(valueAtom).value,
-      setValue: async (value: Value) => {
-        await defer()
-        setSelf({ type: 'setValueAsync', payload: value })
-      },
-    }),
+    (_get, { setSelf }) => {
+      const getValue = () => setSelf({ type: 'getValue' }) as Value
+      const store: Store<Value> = {
+        proxyState: createProxyState(getValue(), () => store),
+        getValue,
+        setValue: (value: Value) =>
+          setSelf({ type: 'setValue', payload: value }) as void,
+      }
+      return store
+    },
     (get, set, action) => {
-      if (action.type === 'setValueAsync') {
+      if (action.type === 'setValue') {
         set(valueAtom, { value: action.payload })
-      } else if (action.type === 'mount') {
-        const store = get(storeAtom)
-        store.setValue = (value: Value) => {
-          set(valueAtom, { value })
-        }
+      } else if (action.type === 'getValue') {
+        return get(valueAtom).value
```

Comment:
is this to avoid potential pitfalls with accidentally subscribing valueAtom to Jotai's dependency map?

Reply:
Yes.